### ISSUE TITLE
Multiple Youtube recipes are failing due to (long title)!

### ISSUE THREAD

**Issue by RavanJAltaie** [openzim/zim-requests#1102]
I've found the below YouTube recipes failing due to (Title is too long!) issue:
1- https://farm.openzim.org/recipes/avanti-biomolecules-everyday-life
2- https://farm.openzim.org/recipes/avanti-continuity-differentiability
3- https://farm.openzim.org/recipes/avanti-magnetic-effect-electric-current
4- https://farm.openzim.org/recipes/avanti-mechanical-properties-of-fluids
5- https://farm.openzim.org/recipes/avanti-mechanical-properties-of-solids
6- https://farm.openzim.org/recipes/avanti-principles-mathematical-induction
7- https://farm.openzim.org/recipes/avanti-permutations-combinations
8- https://farm.openzim.org/recipes/avanti-sets-relations-functions
9- https://farm.openzim.org/recipes/avanti-thermal-properties-of-matter

**Comment by RavanJAltaie** [comment]
Fixing is on progress.

**Comment by RavanJAltaie** [comment]
@benoit74 All recipes above have been fixed for long title, but they are still failing due to:
"Unable to connect to Youtube API v3. check `API_KEY`

**Comment by benoit74** [comment]
Of course, I had a look at three of them and they have all failed when the issue on youtube scraper was still not solved: https://github.com/kiwix/operations/issues/217 ; probably the case for all of them (I'm too lazy to open them all).

You need to request them again now that issue is solved.

**Comment by benoit74** [comment]
I had to fix myself all these recipes title:

<google-sheets-html-origin><style type="text/css"></style>
avanti-basic-concepts-of-chemistry
--
avanti-biomolecules-everyday-life
avanti-continuity-differentiability
avanti-inverse-trigonometry
avanti-magnetic-effect-electric-current
avanti-mechanical-properties-of-fluids
avanti-mechanical-properties-of-solids
avanti-periodic-properties
avanti-permutations-combinations
avanti-principles-mathematical-induction
avanti-rotation
avanti-sets-relations-functions
avanti-thermal-properties-of-matter
avanti-thermodynamics
avanti-thermodynamics-and-kinetic-theory
avanti-vector-and-calculus
canadian_prepper_bugoutconcepts_en
canadian_prepper_preppingfood_en
canadian_prepper_winterprepping_en
litterature-audiobooks-poetry_fr
oer4schools
premiers_pas_avec_python_fr

@RavanJAltaie please fix following ones (I can't read arabic, you are probably well better served on these ones):
madrasa_ar_electeng
madrasa_ar_magnets
madrasa_ar_songs

Regarding avanti recipes, I think my fixes are only short-term. I don't get why we put the description of the content in the title, and why all these titles are in English if these ZIMs are in Hindi. From my perspective we should have a short title like "Avanti courses" in Hindi. And the description should be clear enough to know what the ZIM is about. Here it looks like it is the same on all ZIMs and it says something like "Avanti Gurukul has created the Sankalp program for government school children." which explains only little about the ZIM content.

I also think that playlist ID of https://farm.openzim.org/recipes/avanti-thermodynamics is wrong. It is the same as https://farm.openzim.org/recipes/avanti-periodic-properties.

**Comment by RavanJAltaie** [comment]
@benoit74 madrasa recipes are fixed.
regarding avanti recipes, I've found 6 or 7 recipes failed due to invalid playlist ID.
I'll check and let you know here once fixed.

**Comment by benoit74** [comment]
OK, thank you

**Comment by benoit74** [comment]
Why did we renamed the ZIM/book for many avanti recipes?

This has created many duplicate ZIMs in the library.

See for instance: https://library.kiwix.org/#lang=&q=magnetic

It was named "avanti-magnetic-effect-electric-current" in March 2024: https://farm.openzim.org/pipeline/7700ccd9-5811-4153-b12b-8f302a08eafb
And is now named "avanti-magnetic-effect": https://farm.openzim.org/pipeline/7ebe8f12-7d1f-408a-a159-b1a8b93be1bf

I see no reason to do it + this shouldn't be done without proper coordination to avoid duplicate ZIMs in the library. The ZIM name is a very important identifier for some of our users as well, since this is what allow them to be "notified" when an update is done to a ZIM which interest them.

How should we clean this? Is there a good reason to move to the new name, or should we go back to previous one?

**Comment by RavanJAltaie** [comment]
@benoit74 the reason was to synchronize the title with the recipe name for a better organization, I checked that were no similar recipe name to avoid dublicating.

**Comment by benoit74** [comment]
I don't mind you've changed title and recipe name. My concern is that you've also change the ZIM name and this causes duplicates in the library. How do we clean this?

**Comment by benoit74** [comment]
We've agreed about two weeks ago that @RavanJAltaie will send me the list of old/new ZIM names so that I can rename old ZIMs to new name.

**Comment by RavanJAltaie** [comment]
@benoit74 
Please find the files to fix in the link below
https://drive.farm.openzim.org/avanti/Duplicated%20Avanti%20Zim%20Files.xlsx

**Comment by benoit74** [comment]
I've renamed files mostly as suggested in the Excel sheet (there is duplicates and I've found few which were missing).
I think that this issue is now closed.
